client/pyre.py

```diff
@@ -873,6 +873,60 @@ def persistent(context: click.Context, no_watchman: bool) -> int:
         )
 
 
+@pyre.command()
+@click.option("--no-watchman", is_flag=True, default=False, hidden=True)
+@click.pass_context
+def pysa_language_server(context: click.Context, no_watchman: bool) -> int:
+    """
+    Entry point for IDE integration to Pysa. Communicates with a Pysa server using
+    the Language Server Protocol, accepts input from stdin and writing diagnostics
+    and responses from the Pysa server to stdout.
+    """
+    command_argument: command_arguments.CommandArguments = context.obj["arguments"]
+    configuration = configuration_module.create_configuration(
+        command_argument, Path(".")
+    )
+    if configuration.use_command_v2:
+        log.start_logging_to_directory(
+            # Always log to file regardless of whether `-n` is given
+            noninteractive=False,
+            log_directory=configuration.log_directory,
+        )
+        return v2.pysa_server.run(
+            configuration,
+            command_arguments.StartArguments(
+                changed_files_path=command_argument.changed_files_path,
+                debug=command_argument.debug,
+                enable_memory_profiling=command_argument.enable_memory_profiling,
+                enable_profiling=command_argument.enable_profiling,
+                load_initial_state_from=command_argument.load_initial_state_from,
+                log_identifier=command_argument.log_identifier,
+                logging_sections=command_argument.logging_sections,
+                no_saved_state=command_argument.no_saved_state,
+                no_watchman=no_watchman,
+                noninteractive=command_argument.noninteractive,
+                save_initial_state_to=command_argument.save_initial_state_to,
+                saved_state_project=command_argument.saved_state_project,
+                sequential=command_argument.sequential,
+                show_error_traces=command_argument.show_error_traces,
+                store_type_check_resolution=False,
+                terminal=False,
+                wait_on_initialization=True,
+            ),
+        )
+    else:
+        return run_pyre_command(
```

Comment:
This looks like it's still calling the old pyre persistent command. Should this be here?

Reply:
I am confused as to whether or not anything should execute in case of "v1" command call. Should we only allow v2? If so, I will just delete that else statement.

Reply:
> I am confused as to whether or not anything should execute in case of "v1" command call.  Should we only allow v2?

Yes. V1 is considered legacy code these days. We are still in the process of migrating away from it. New code can use V2 directly.